### PR TITLE
feat(ext-plugin): implement the http-req-call protocol

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/ext-plugin/init.lua [hunk 1]
```diff
@@ -152,6 +165,80 @@ end
...
+        PUT = a6_method.PUT,
+        DELETE = a6_method.DELETE,
+        MKCOL = a6_method.MKCOL,
+        COPY = a6_method.COPY,
+        MOVE = a6_method.MOVE,
+        OPTIONS = a6_method.OPTIONS,
+        PROPFIND = a6_method.PROPFIND,
+        PROPPATCH = a6_method.PROPPATCH,
+        LOCK = a6_method.LOCK,
+        UNLOCK = a6_method.UNLOCK,
```
[the remark this answers is not in the document]
Comment (spacewander) [reply apisix/plugins/ext-plugin/init.lua:200]: Those fields are copied from https://github.com/openresty/lua-nginx-module#http-method-constants

File: apisix/plugins/ext-plugin/init.lua [hunk 2]
```diff
@@ -152,6 +165,80 @@ end
...
+do
+    local count = 0
+    local MAX_COUNT = lshift(1, 22)
+
+    function generate_id()
+        local wid = worker_id()
+        local id = lshift(wid, 22) + count
+        count = count + 1
+        if count == MAX_COUNT then
+            count = 0
```
Comment (tokers) [inline apisix/plugins/ext-plugin/init.lua:178]: Does the id rotate will influence the correctness?

File: apisix/plugins/ext-plugin/init.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (membphis) [reply apisix/plugins/ext-plugin/init.lua:200]: got it, many thanks for explaining

**Review by membphis — APPROVED** [review]

File: apisix/plugins/ext-plugin/init.lua (hunk 2 above)
Comment (spacewander) [reply apisix/plugins/ext-plugin/init.lua:178]: The id here is only used for logging purpose. In the given time window, it is rare to have two repeated id so we can distinguish them.

Comment (tokers) [reply apisix/plugins/ext-plugin/init.lua:178]: OK.

**Review by tokers — APPROVED** [review]

**Pull request merged**